app/src/main/java/com/example/plantsapp/data/contentprovider/PlantsContentProvider.kt

```diff
@@ -0,0 +1,100 @@
+@file:Suppress("WildcardImport", "NoWildcardImports")
+package com.example.plantsapp.data.contentprovider
+
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import com.example.plantsapp.di.module.FirebaseQualifier
+import com.example.plantsapp.domain.model.Plant
+import com.example.plantsapp.domain.repository.PlantsRepository
+import com.example.plantstatscontract.*
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.runBlocking
+
+class PlantsContentProvider : ContentProvider() {
+
+    @EntryPoint
+    @InstallIn(SingletonComponent::class)
+    interface PlantsContentProviderEntryPoint {
+        @FirebaseQualifier
+        fun plantsRepository(): PlantsRepository
+    }
+
+    private val plantsRepository: PlantsRepository by lazy {
+        val appContext = context?.applicationContext
+            ?: throw IllegalStateException("Unable to get applicationContext")
+
+        val hiltEntryPoint = EntryPointAccessors.fromApplication(
+            appContext,
+            PlantsContentProviderEntryPoint::class.java
+        )
+        hiltEntryPoint.plantsRepository()
```

Comment:
I'm not sure -so we can discuss this, but how about make hiltentrypoint as latent inside of ContentProvider class, and instantiate it inside of onCreate? 
and plant repository will be same - by lazy{ hiltEntryPoint.plantsRepository}